src/Core/Scheduler/Providers/SingleImageProvider.js

```diff
@@ -0,0 +1,52 @@
+import { Vector3 } from 'three';
+import Extent from '../../Geographic/Extent';
+import OGCWebServiceHelper from './OGCWebServiceHelper';
+
+export default {
+    preprocessDataLayer(layer) {
+        if (!layer.extent) {
+            throw new Error('layer.extent is required');
+        }
+
+        if (!(layer.extent instanceof Extent)) {
+            layer.extent = new Extent(layer.projection, layer.extent);
+        }
+
+        layer.options = {};
+    },
+
+    tileInsideLimit(tile, layer) {
+        // Only fetch level 0 (full res texture): node with level > 0 will inherit the
+        // texture from their parent
+        return tile.level === 0 && layer.extent.intersect(tile.extent);
+    },
+
+    getColorTexture(tile, layer) {
+        if (!this.tileInsideLimit(tile, layer)) {
+            return Promise.reject(`Tile '${tile}' is outside layer bbox ${layer.extent}`);
```

Comment:
This will produce something like 

> Tile '[object Object]' is outside layer bbox '[object Object]'

Also, usually, people reject with an Error object, not a single string, which could allow you to attach tile and layer to this error object maybe? Not sure what is best here.